packages/docs-website/docs/protocol-docs/client-specification/json-rpc-api.md

```diff
@@ -0,0 +1,420 @@
+
+### Response
+
+// TODO: (HIGH): What about the `SingleChannelResult` and `MultipleChannelResult` documented in the `server-wallet` package?
+
+This method returns a `ChannelResult` as part of the JSON RPC Response, which is an object with the following fields:
+
+| Name                    | Type              | Description                                |
+| ----------------------- | ----------------- | ------------------------------------------ |
+| participants            | `Participant[]`   | Identifying information members of channel |
+| allocations             | `Allocation[]`    | Array of funding amounts for participants  |
+| appData                 | string            | Encoded initial state of app               |
+| appDefinition           | string            | Address of contract governing the channel  |
+| channelId               | string            | Unique channel identifier                  |
+| status                  | `ChannelStatus`   | Current status of channel                  |
+| turnNum                 | number            | Channel nonce                              |
+| challengeExpirationTime | number (optional) | Time current challenge on channel elapses  |
+
+where the channel status can be one of:
+
+TODO: (HIGH) Decided on finalized statuses. What is opening vs. proposed vs. funding vs. running? Is the channel thrown back into the `funding` state when it runs out of money?
+
+- `proposed`: wallet has stored a channel, but no states are signed
+- `opening`: channel has been joined, but is not properly funded
+- `funding`: ???
+- `running`: channel is ready to use
+- `challenging`: channel is in an ongoing dispute
+- `responding`: channel dispute is ongoing, and it is your turn to create a channel update (must send `UpdateChannel` request)
+- `closing`: channel cannot be used, but funds are still locked
+- `closed`: funds have been released from channel
+
+Generally, the `ChannelResult` type is returned when an update to the channel state was made.
+
+### Errors
+
+| Code | Message                | Description                           |
+| ---- | ---------------------- | ------------------------------------- |
+| 1000 | SigningAddressNotFound | Unable to find expected ephemeral key |
+| 1001 | InvalidAppDefinition   | App definition not valid address      |
+| 1002 | UnsupportedToken       | Token in allocations not supported    |
+
+TODO: (HIGH) Verify these are all of the expected errors (what if request times out? what if some participant refuses to join the channel?)
+
+## ApproveBudgetAndFund
+
+TODO: (HIGH) Better description of the actual usage of this rpc call. Is this essentially a collateral request? How much of an option do people have to reject these requests? Is this rejection exposed through the wallet, or would it have to be "undone" by some other series of wallet calls? Is this how you install apps in a ledger channel?
```

Comment:
Budgets currently serve a dual purpose (I would say).

1. They are essentially a permission, set by the user, allowing the wallet to silently open, _virtually fund_ , and close channels for a certain domain (without prompting the user), up to the given amounts (one for sending, one for receiving). The user has the chance to reject the budget via a wallet popup. If the budget runs out, they would need to re-approve another one. 
2. The budget is typically serviced by a ledger channel with the hub, so this _will_ require collateral from the hub. The ledger channel allocations typically match the budget amounts (my maximum send is my balance in the ledger channel, my maximum receive is the hub's balance in the ledger channel). 

See also [here](https://web3torrent.statechannels.org/budgets) and [here](https://www.notion.so/web3Torrent-FAQ-9f384d9dbadc4828aa81e14fcc360466).

Reply:
I've removed the documentation here, since it is a browser-specific method

Reply:
But thanks for the detailed explainer!